src/components/fields/contact-field/contact-field.tsx

```diff
@@ -107,11 +109,22 @@ export const ContactField = (props: IGenericFieldProps<IContactFieldSchema>) =>
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [validation, defaultCountry]);
 
-	// handle changes in value through reset
+	// handles changes in value through reset or defaultValues
+	// adds prefix if it is not specified
 	useEffect(() => {
-		const { number } = PhoneHelper.getParsedPhoneNumber(value || "");
-		setStateValue(number);
-	}, [value]);
+		if (fixedCountry !== undefined) {
+			const { prefix, number } = PhoneHelper.getParsedPhoneNumber(value || "");
+			const prefixWithoutPlus = prefix.replace(/\+/g, "");
+			const schemaPrefix = getPrefixFromCountry(countryValue || "Singapore");
+			if (!fixedCountry || (fixedCountry && prefixWithoutPlus === schemaPrefix)) {
+				const fieldPrefix = prefix || schemaPrefix;
+				handleChange({ countryCode: fieldPrefix, number });
+			} else {
+				handleChange({ countryCode: "", number: "" });
```

Comment:
if fixedCountry === false, would this end up changing the default value to `""`?

Reply:
if fixedCountry===false, it will pass the default value into handleChange in line 121